src/analyzer/analyzerthread.cpp

```diff
@@ -248,45 +251,46 @@ AnalyzerThread::AnalysisResult AnalyzerThread::analyzeAudioSource(
         DEBUG_ASSERT(!chunkFrameRange.empty());
 
         // Request the next chunk of audio data
-        const auto readableSampleFrames =
-                audioSourceProxy.readSampleFrames(
-                        mixxx::WritableSampleFrames(
-                                chunkFrameRange,
-                                mixxx::SampleBuffer::WritableSlice(m_sampleBuffer)));
+        auto writableSampleFrames =
+                mixxx::WritableSampleFrames(
+                        chunkFrameRange,
+                        mixxx::SampleBuffer::WritableSlice(m_sampleBuffer));
+
+        // Request the next chunk of audio data
+        auto readableSampleFrames =
+                audioSourceProxy.readSampleFrames(writableSampleFrames);
         // The returned range fits into the requested range
         DEBUG_ASSERT(readableSampleFrames.frameIndexRange() <= chunkFrameRange);
 
         // Sometimes the duration of the audio source is inaccurate and adjusted
         // while reading. We need to adjust all frame ranges to reflect this new
         // situation by restoring all invariants and consistency requirements!
 
-        // Shrink the original range of the current chunks to the actual available
-        // range.
-        chunkFrameRange = intersect(chunkFrameRange, audioSourceProxy.frameIndexRange());
-        // The audio data that has just been read should still fit into the adjusted
-        // chunk range.
-        DEBUG_ASSERT(readableSampleFrames.frameIndexRange() <= chunkFrameRange);
-
-        // We also need to adjust the remaining frame range for the next requests.
-        remainingFrameRange = intersect(remainingFrameRange, audioSourceProxy.frameIndexRange());
-        // Currently the range will never grow, but lets also account for this case
-        // that might become relevant in the future.
-        VERIFY_OR_DEBUG_ASSERT(remainingFrameRange.empty() ||
-                remainingFrameRange.end() == audioSourceProxy.frameIndexRange().end()) {
-            if (chunkFrameRange.length() < mixxx::kAnalysisFramesPerChunk) {
-                // If we have read an incomplete chunk while the range has grown
-                // we need to discard the read results and re-read the current
-                // chunk!
-                remainingFrameRange = span(remainingFrameRange, chunkFrameRange);
-                continue;
-            }
-            DEBUG_ASSERT(remainingFrameRange.end() < audioSourceProxy.frameIndexRange().end());
-            kLogger.warning()
-                    << "Unexpected growth of the audio source while reading"
-                    << mixxx::IndexRange::forward(
-                            remainingFrameRange.end(), audioSourceProxy.frameIndexRange().end());
-            remainingFrameRange.growBack(
-                    audioSourceProxy.frameIndexRange().end() - remainingFrameRange.end());
+        if (readableSampleFrames.frameIndexRange().start() > chunkFrameRange.start()) {
```

Comment:
Missing samples at the front and missing samples at the back of the requested range are independent and should be handled subsequently, not in mutual exclusive branches. Missing samples at the front are always decoding errors while missing samples at the back of the range could either be decoding errors or an inaccurate stream size.